code-search/code-search.ipynb

```diff
@@ -0,0 +1,609 @@
+        "# model. Please visit https://huggingface.co/jinaai/jina-embeddings-v2-base-code\n",
+        "# to accept the rules and generate the access token in your account settings:\n",
+        "# https://huggingface.co/settings/tokens\n",
+        "\n",
+        "HF_TOKEN = \"THIS_IS_YOUR_TOKEN\""
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Once the token is ready, we can pass the code snippets through the second model."
+      ],
+      "metadata": {
+        "id": "2yk2CtOtTzF8"
+      }
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "JcPlZxZmi4Ag"
+      },
+      "outputs": [],
+      "source": [
+        "code_model = SentenceTransformer(\n",
+        "    \"jinaai/jina-embeddings-v2-base-code\",\n",
+        "    token=HF_TOKEN,\n",
+        "    trust_remote_code=True\n",
+        ")\n",
+        "code_model.max_seq_length = 8192  # increase the context length window\n",
+        "code_embeddings = code_model.encode(\n",
+        "    code_snippets, batch_size=4, show_progress_bar=True,\n",
+        ")\n",
+        "code_embeddings.shape"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Created embeddings have to be indexed in a Qdrant collection. For that, we need a running instance. The easiest way is to deploy it using the [Qdrant Cloud](https://cloud.qdrant.io/). There is a free tier 1GB cluster available, but you can alternatively use [a local Docker container](https://qdrant.tech/documentation/quick-start/), but running it in Google Colab might require installing Docker first."
+      ],
+      "metadata": {
+        "id": "zCyTLRV_T5jb"
+      }
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "Cw_gxsdciMgu"
+      },
+      "outputs": [],
+      "source": [
+        "QDRANT_URL = \"https://my-cluster.cloud.qdrant.io:6333\" # http://localhost:6333 for local instance\n",
+        "QDRANT_API_KEY = \"THIS_IS_YOUR_API_KEY\" # None for local instance"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "9gtJ79JylztO"
+      },
+      "outputs": [],
+      "source": [
+        "from qdrant_client import QdrantClient, models\n",
+        "\n",
+        "client = QdrantClient(QDRANT_URL, api_key=QDRANT_API_KEY)\n",
+        "client.create_collection(\n",
+        "    \"qdrant-sources\",\n",
+        "    vectors_config={\n",
+        "        \"text\": models.VectorParams(\n",
+        "            size=nlp_embeddings.shape[1],\n",
+        "            distance=models.Distance.COSINE,\n",
+        "        ),\n",
+        "        \"code\": models.VectorParams(\n",
+        "            size=code_embeddings.shape[1],\n",
+        "            distance=models.Distance.COSINE,\n",
+        "        ),\n",
```

Comment:
I'd love it if we can mention / call out that this is what Qdrant calls Named Vectors